### PR TITLE
Return build info under `/loki/api/v1/status/buildinfo`.

### LINKED ISSUE #3221: HTTP API Endpoint to get Loki version
**Is your feature request related to a problem? Please describe.**
Prometheus has this endpoint: ```GET /api/v1/status/buildinfo``` which will return the version of Prometheus. I don't see any way using Loki's HTTP API to figure out what version of Loki I'm talking to.

**Describe the solution you'd like**
```GET /api/v1/status/buildinfo``` or alternative for Loki.

**Additional context**
I'm using Grafana Cloud, so I don't have any access or control over the Loki server. I don't know what version of Loki I'm using, I'd like to know so that I know when I can use new features that become available.

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
This changes introduces the `/version` endpoint for all components. It returns the build info in JSON. This should ease debugging.

**Which issue(s) this PR fixes**:
Fixes #3221

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Review by dannykopping — APPROVED** [review]
LGTM :+1:

File: docs/sources/api/_index.md [hunk 1]
```diff
@@ -823,6 +824,12 @@ and the current are returned. A value of `defaults` returns the default configur
 
 In microservices mode, the `/config` endpoint is exposed by all components.
 
+## `GET /version`
+
+`/version` exposes the build information in a JSON object. The fields are `version`, `revision ``build_date`, `build_user`, and `revision`.
```
Comment (kavirajk) [inline docs/sources/api/_index.md:829]: Revision is repeated twice!

**Review by kavirajk — APPROVED** [review]
LGTM 🎉. One small comment!

File: docs/sources/api/_index.md (hunk 1 above)
Comment (jeschkies) [reply docs/sources/api/_index.md:829]: Good catch.

**Review by owen-d — APPROVED** [review]
Looks good, but can we mimic the same path that Prometheus uses (`GET /loki/api/v1/status/buildinfo` in our case)?
https://prometheus.io/docs/prometheus/latest/querying/api/#build-information

Nit: we can probably use the same version struct Prometheus uses.

File: pkg/util/build/build.go [hunk 2]
```diff
@@ -20,4 +21,5 @@ func init() {
 	version.Branch = Branch
 	version.BuildUser = BuildUser
 	version.BuildDate = BuildDate
+	version.GoVersion = "???"
```
Comment (jeschkies) [inline pkg/util/build/build.go:24]: How can I access the Go version?

**Review by KMiller-Grafana — APPROVED** [review]
Docs look good to me.

File: pkg/util/build/build.go (hunk 2 above)
Comment (owen-d) [reply pkg/util/build/build.go:24]: You can use the `runtime` package: https://golang.org/pkg/runtime/#Version

Comment (owen-d) [reply pkg/util/build/build.go:24]: That's how Prometheus does it as well: https://github.com/prometheus/common/blob/main/version/info.go#L33

**Pull request merged**